### PR TITLE
Added generic Htx test

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Added generic Htx test

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
Job Log:

[stdout] htx_package/
[stdout] htx_package/htx.tar.gz
[stdout] htx_package/installer.sh
[stdout] htx_package/uninstaller.sh
Command 'tar --touch -xvzf htx_package.tar.gz' finished with 0 after 0.0438950061798s
[stdout] htx_package/os_build
Running './installer.sh -f'
[stdout] 
[stdout] Your Install System:
[stdout] Distro:RedHat
[stdout] Arch:ppc64le
[stdout] 
[stdout] Installing HTX in default path:/usr/lpp/htx...
[stdout] 
[stdout] Creating 'htx' user...[OK]
[stdout] 
[stdout] Extracting htx.tar.gz to /usr/lpp...[OK]
[stdout] 
[stdout] Finishing Installation...[OK]
[stdout] 
[stdout] Copying HTX Binaries...[OK]
[stdout] 
[stdout] Creating htx.d.service file... [OK]
[stdout] 
[stdout] Adding HTX Daemon(htxd) as service...[OK]
[stdout] 
[stdout] Checking HTX daemon status...[Running]
[stdout] 
[stdout] HTX install completed..Now exiting.
Command './installer.sh -f' finished with 0 after 6.22520303726s
Starting the HTX Deamon
Running '/usr/lpp/htx/etc/scripts/htxd_run'
[stdout] [Thu Nov 23 00:40:03 CST 2017]: exported HTX_HOME_DIR=/usr/lpp/htx/
[stdout] [Thu Nov 23 00:40:03 CST 2017]: exported PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin:/usr/lpp/htx/:/usr/lpp/htx//etc/scripts:/usr/lpp/htx//etc:/usr/lpp/htx//etc/methods:/usr/lpp/htx//bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/ucb:/usr/bin/X11:/etc:/test/tools:/usr/local/bin:/usr/lpp/htx//test/tools:.:/usr/lpp/htx//bin/:/usr/lpp/htx//etc//scripts_stx:/usr/lpp/htx//etc/scripts/:/usr/lpp/htx/:/usr/lpp/htx//etc/scripts:/usr/lpp/htx//etc:/usr/lpp/htx//etc/methods:/usr/lpp/htx//bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/ucb:/usr/bin/X11:/etc:/test/tools:/usr/local/bin:/usr/lpp/htx//test/tools:.
[stdout] [Thu Nov 23 00:40:03 CST 2017]: 1st HTX message.
[stdout] [Thu Nov 23 00:40:03 CST 2017]: exporting PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin:/usr/lpp/htx/:/usr/lpp/htx//etc/scripts:/usr/lpp/htx//etc:/usr/lpp/htx//etc/methods:/usr/lpp/htx//bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/ucb:/usr/bin/X11:/etc:/test/tools:/usr/local/bin:/usr/lpp/htx//test/tools:.:/usr/lpp/htx//bin/:/usr/lpp/htx//etc//scripts_stx:/usr/lpp/htx//etc/scripts/:/usr/lpp/htx/:/usr/lpp/htx//etc/scripts:/usr/lpp/htx//etc:/usr/lpp/htx//etc/methods:/usr/lpp/htx//bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/ucb:/usr/bin/X11:/etc:/test/tools:/usr/local/bin:/usr/lpp/htx//test/tools:.
[stderr] /usr/lpp/htx/etc/scripts/htx_setup.sh: line 70: [: too many arguments
[stdout] [Thu Nov 23 00:40:04 CST 2017]: exporting HTX_RELEASE=htxrhel72le
[stdout] [Thu Nov 23 00:40:04 CST 2017]: Setting ulimit -s(stack size) to 8MB. return=0
[stdout] [Thu Nov 23 00:40:04 CST 2017]: Setting ulimit -c(core limit) to 'unlimited'. return=0
[stdout] [Thu Nov 23 00:40:04 CST 2017]: Setting ulimit -v(virtual memory)  to 'unlimited'. return=0
[stdout] [Thu Nov 23 00:40:04 CST 2017]: Setting ulimit -n(# open files) to '1024'. return=0
[stdout] [Thu Nov 23 00:40:04 CST 2017]: HTXD_STARTUP_MODE was 'ON'. Turning it OFF and returning.
[stdout] Starting HTX Daemon... 
[stdout] /usr/lpp/htx//bin//htxd > /tmp//daemon_log 2>&1
[stderr] ERROR: while binding connection. Exiting... errno = 98, return_code = 17: Address already in use
[stderr] ERROR: while binding connection. Exiting... errno = 98, return_code = 18: Address already in use
Command '/usr/lpp/htx/etc/scripts/htxd_run' finished with 0 after 3.00578713417s
selecting the mdt file
Running 'htxcmdline -select -mdt mdt.mem'
[stdout] 
Command 'htxcmdline -select -mdt mdt.mem' finished with 0 after 76.2377269268s
[stdout] ######################## Result Starts Here ################################
[stdout] ECG (/usr/lpp/htx/mdt/mdt.mem) is selected
[stdout] ######################### Result Ends Here #################################
Activating the mdt.mem
Running 'htxcmdline -activate -mdt mdt.mem'
[stdout] 
Command 'htxcmdline -activate -mdt mdt.mem' finished with 0 after 0.00191593170166s
[stdout] ######################## Result Starts Here ################################
[stdout] Currently running ECG/MDT : /usr/lpp/htx/mdt/mdt.mem
[stdout] ===========================
[stdout] State   Dev     Adapt Desc   Device Desc         Slot Port
[stdout] -----------------------------------------------------------
[stdout] ACTIVE  mem     64bit        memory              NA           
[stdout] ######################### Result Ends Here #################################
Running the HTX 
Running 'htxcmdline -run  -mdt mdt.mem'
Command 'htxcmdline -run  -mdt mdt.mem' finished with 0 after 10.5569229126s
[stdout] 
[stdout] ######################## Result Starts Here ################################
[stdout] ECG (/usr/lpp/htx/mdt/mdt.mem) Activated.
[stdout] ######################### Result Ends Here #################################
HTX Error logs
Running 'htxcmdline -geterrlog'
[stdout] 
Command 'htxcmdline -geterrlog' finished with 0 after 0.00180602073669s
[stdout] ######################## Result Starts Here ################################
[stdout] Currently running ECG/MDT : /usr/lpp/htx/mdt/mdt.mem
[stdout] ===========================
[stdout] file </tmp/htxerr> is empty
[stdout] ######################### Result Ends Here #################################
Running 'htxcmdline -query  -mdt mdt.mem'
[stdout] 
Command 'htxcmdline -query  -mdt mdt.mem' finished with 0 after 0.00175499916077s
[stdout] ######################## Result Starts Here ################################
[stdout] Currently running ECG/MDT : /usr/lpp/htx/mdt/mdt.mem
[stdout] ===========================
[stdout] --------------------------------------------------------------------------------
[stdout]  Device   ST  ACTIVE COE Last Update  Count Stanza Last Error   Slot/  Num_errs
[stdout]              SUSPEND SOE day Time                  Day  Time    Port
[stdout] --------------------------------------------------------------------------------
[stdout] mem       RN  ACTIVE SOE 327 00:41:28 0        0   NA   NA       NA            0
[stdout] ######################### Result Ends Here #################################
shutting down the mdt.mem 
Running 'htxcmdline -shutdown -mdt mdt.mem'
[stdout] 
Command 'htxcmdline -shutdown -mdt mdt.mem' finished with 0 after 14.0720820427s
[stdout] ######################## Result Starts Here ################################
[stdout] ECG (/usr/lpp/htx/mdt/mdt.mem) shutdown successfully
[stdout] ######################### Result Ends Here #################################
Running '/etc/init.d/htx.d status'
Command '/etc/init.d/htx.d status' finished with 0 after 0.0445261001587s
[stdout] htx daemon is running
Running '/usr/lpp/htx/etc/scripts/htxd_shutdown'
Command '/usr/lpp/htx/etc/scripts/htxd_shutdown' finished with 0 after 0.0712199211121s
[stdout] htxd is already exited...
Running './uninstaller.sh -f'
[stdout] 
[stdout] Checking HTX daemon Status...[Running]
[stdout] 
[stdout] Checking HTX tests Status...[Not Activated]
[stdout] 
[stdout] HTX daemon Stopped Successfully...
[stdout] 
[stdout] Removing HTX Daemon(htxd) as service...[OK]
[stdout] 
[stdout] Removing binaries...[OK]
[stdout] 
[stdout] Removing HTX from installed path: /usr/lpp/htx...[OK]
[stdout] 
[stdout] Removing htx daemon...[OK]
[stdout] 
[stdout] Deleting user 'htx'...[OK]
[stdout] 
[stdout] Deleting file /var/log/htx_install_path...
[stdout] 
[stdout] Uninstall finished... Exiting.
Command './uninstaller.sh -f' finished with 0 after 4.87397003174s
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
PASS 1-htx_test.py:HtxTest.test

Not logging /proc/pci (file does not exist)
Test results available in /root/avocado-fvt-wrapper/results/job-2017-11-23T00.39-ff7cffa
[root@System ~]#

File: generic/htx_test.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+from avocado.utils import distro
+from avocado.utils import cpu
+
+
+class HtxTest(Test):
+
+    """
+    HTX [Hardware Test eXecutive] is a test tool suite. The goal of HTX is to
+    stress test the system by exercising all hardware components concurrently
+    in order to uncover any hardware design flaws and hardware hardware or
```
Comment (nasastry) [inline generic/htx_test.py]: s/hardware hardware/hardware-hardware/

File: generic/htx_test.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        archive.extract(tarball, self.teststmpdir)
+        htx_path = os.path.join(self.teststmpdir, "HTX-master")
+        os.chdir(htx_path)
+
+        build.make(htx_path, extra_args='all')
+        build.make(htx_path, extra_args='tar')
+        process.run('tar --touch -xvzf htx_package.tar.gz')
+        os.chdir('htx_package')
+        if process.system('./installer.sh -f'):
+            self.fail("htx installation please check log about any issue ")
```
Comment (nasastry) [inline generic/htx_test.py]: Fail message can be better.

File: generic/htx_test.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    :params time_limit: how much time(hours) you want to run this stress.
+    :smt_change : if user want to change smt value as well while running test
+    """
+
+    def setUp(self):
+        """
+        Build 'HTX'.
+        """
+        if 'ppc64' not in process.system_output('uname -a', shell=True):
+            self.cancel("Platform does not supports")
```
Comment (nasastry) [inline generic/htx_test.py]: Cancel message can be better.

File: generic/htx_test.py (hunk 1 above)
Comment (PraveenPenguin) [reply generic/htx_test.py]: sure

File: generic/htx_test.py (hunk 3 above)
Comment (PraveenPenguin) [reply generic/htx_test.py]: will do , thanks for suggestion

File: generic/htx_test.py (hunk 2 above)
Comment (PraveenPenguin) [reply generic/htx_test.py]: sure

**Comment by PraveenPenguin** [comment]
@nasastry done the changes please review

**Comment by PraveenPenguin** [comment]
@nasastry thanks for ACK , merging as one ACK is there

**Pull request merged**